### DISCUSSION TITLE
Shared folders subdirectories do not sync?

### DISCUSSION THREAD

**Discussion by nijatismayilzada** [abraunegg/onedrive#1665] [Q&A] [answered]
Hi,

I have been trying to use this client to sync [this](https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#shared-folders-onedrive-personal) type of folders. Folder contents get synced, the client creates subfolders too, but all subfolders are empty. In the logs, I can see some stuff getting ignored with this log:

`Remote change discarded - not in sync scope`

Any chance that this is a known issue? 

Thanks.

**Comment by abraunegg** [comment, marked as answer]
@nijatismayilzada 
This is not a known issue.

A few questions:
* What application version are you using? Use `onedrive --version`
* What is your application configuration? Use `onedrive --display-config`
* How are you starting the application - what is the exact command you are using?
* What are your debug logs showing as to what is occuring - add `--verbose --verbose` to how you are starting the application and provide this output

**Edit:**
* Account Type is OneDrive Personal
* Add shared folder to OneDrive account via 'Add to my OneDrive' which adds as a link
* Sync data locally

**Online State:**
![Capture](https://user-images.githubusercontent.com/4956234/138574269-f4b0194c-c7d6-4a85-87e7-65bc06762a27.JPG)

![Capture2](https://user-images.githubusercontent.com/4956234/138574270-047a6880-ec32-4e9a-9eee-9a11c19520f2.JPG)

**Whilst syncing:**
```
...
Syncing this OneDrive Personal Shared Folder: 3A
Applying changes of Path ID: BC7D88EC1F539DCF!107
Number of items from OneDrive to process: 4
Downloading file 3A/Remove Ubuntu Package.docx ... 
Downloading 100% |oooooooooooooooooooooooooooooooooooooooo| DONE IN 00:00:01                                                                                                                  
done.
Downloading file 3A/1/Document2.docx ... done.
...

```

**Local state after sync:**
```
[alex@centos7full 3A]$ tree
.
├── 1
│   └── Document2.docx
└── Remove\ Ubuntu\ Package.docx

1 directory, 2 files
[alex@centos7full 3A]$ 


```

**Comment by robertschulze** [comment]
Actually, I am facing the same issue as @nijatismayilzada - however, very strangely - only for one of my shared folders (I have four of them in my OneDrive). So while two appear to sync fine, one (which in Windows and on the Web is full of files) is containing only empty directories and whatever I tried in terms of (re)configuration, I was not able to make it sync. What could I be missing?

**Reply by abraunegg** [reply]
Just to formalise a response here for future readers:

The issue that @robertschulze was facing was that the OneDrive API was responding with HTML entities in the JSON response, which is a bug in the OneDrive API - raised as an issue here: https://github.com/OneDrive/onedrive-api-docs/issues/1765

This is why, with v2.4.xx code, the respective OneDrive Shared Folder would not sync.

The resolution here was to rename the online folder & shared folder to ensure that this would sync correctly.

When using v2.5.xx, a warning message will now be printed when the OneDrive API responds with HTML entities in the JSON response.
